### ISSUE TITLE
systemd keeps re-adding DNS server of wired connection to /etc/resolv.conf after it's lost its carrier

### ISSUE THREAD

**Issue by WGH-** [systemd/systemd#2375]
The scenario is rather simple: 
- Have wired connection with DHCP enabled
- Connect to it
- Unplug the cable without putting the interface down

systemd will keep re-adding the last "wired DNS server" received via DHCP.

This is especially annoying when you also have a Wi-Fi connectivity. Sometimes the (no longer reachable) "wired DNS server" will be above the wireless ones in the /etc/resolv.conf, and it will delay all DNS lookups in the system substantially.

This is what `networkctl` says about all this (enp0s25 is wired, wlp3s0 is Wi-Fi):

``````
% networkctl                            
IDX LINK             TYPE               OPERATIONAL SETUP     
  1 lo               loopback           carrier     configured
  2 enp0s25          ether              no-carrier  configured
  3 sit0             sit                carrier     configuring
  4 wlp3s0           wlan               routable    configured```
``````

**Comment by poettering** [comment]
/etc/hosts? Do you mean /etc/resolv.conf?

What does you "networkctl status -a" look like, when this happens?

**Comment by WGH-** [comment]
Uhm, yes, `/etc/resolv.conf`, of course.

```
● 2: enp0s25
       Link File: /usr/lib64/systemd/network/99-default.link
    Network File: /etc/systemd/network/wired.network
            Type: ether
           State: no-carrier (configured)
            Path: pci-0000:00:19.0
          Driver: e1000e
          Vendor: Intel Corporation
           Model: Ethernet Connection I218-V
      HW Address: XX:XX:XX:XX:XX:XX (LCFC(HeFei) Electronics Technology Co., Ltd.)
             MTU: 1500
         Address: fe80::XXXX:XXXX:XXXX:XXXX
             DNS: 10.0.0.254
● 4: wlp3s0
       Link File: /usr/lib64/systemd/network/99-default.link
    Network File: /etc/systemd/network/wireless.network
            Type: wlan
           State: routable (configured)
            Path: pci-0000:03:00.0
          Driver: iwlwifi
          Vendor: Intel Corporation
           Model: Wireless 7260 (Wireless-N 7260)
      HW Address: XX:XX:XX:XX:XX:XX (Intel Corporate)
             MTU: 1500
         Address: 192.168.0.154
                  fe80::XXXX:XXXX:XXXX:XXXX
         Gateway: 192.168.0.5 (ASUSTek COMPUTER INC.)
             DNS: 192.168.0.5
```

**Comment by poettering** [comment]
@teg hmm, this suggests networkd keeps the DHCP acquired DNS data around, even if we entered no-carrier. That sounds wrong, no?

**Comment by ssahani** [comment]
@WGH-  Could you please test it with latest git . Not able to reproduce it . I guess this is fixed with #3080

**Comment by poettering** [comment]
Closing as this is most likely fixed with #3080 and no feedback was given.

**Comment by WGH-** [comment]
Sorry for the late reply. I believe it's indeed fixed.
